dsl_parser/framework/parser.py

```diff
@@ -596,8 +612,25 @@ def _extract_element_requirements(element):
             else:
                 if required_type == 'self':
                     required_type = type(element)
-                required_type_elements = context.element_type_to_elements.get(
-                    required_type, [])
+
+                if (
+                    len(requirements) == 1
+                    and requirements[0].predicate == sibling_predicate
+                ):
+                    # If we're only looking for siblings then we can get them
+                    # from the parent rather than looking at every element of
+                    # the type we care about. This cuts the parse time by
+                    # about 7 seconds on sufficiently large blueprints (e.g.
+                    # ones that use the AWS plugin).
+                    required_type_elements = [
+                        child for child in element.parent().children()
+                        if isinstance(child, required_type)]
+                else:
+                    required_type_elements = (
+                        context.element_type_to_elements.get(required_type,
+                                                             [])
+                    )
```

Comment:
hah now that's some weird style, to avoid a `\`

Reply:
It's very pretty!